aggregator/fetchers/factory_test.go

```diff
@@ -0,0 +1,62 @@
+package fetchers
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/aggregator/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewPriceFetcher(t *testing.T) {
+	t.Parallel()
+
+	t.Run("invalid fetcher name should error", func(t *testing.T) {
+		t.Parallel()
+
+		pf, err := NewPriceFetcher("invalid name", nil)
+		assert.Nil(t, pf)
+		assert.Equal(t, errInvalidFetcherName, err)
+	})
+	t.Run("nil responseGetter should error", func(t *testing.T) {
+		t.Parallel()
+
+		pf, err := NewPriceFetcher(binanceName, nil)
+		assert.Nil(t, pf)
+		assert.Equal(t, errNilResponseGetter, err)
+	})
+	t.Run("nil responseGetter should error", func(t *testing.T) {
+		t.Parallel()
+
+		pf, err := NewPriceFetcher(binanceName, nil)
+		assert.Nil(t, pf)
+		assert.Equal(t, errNilResponseGetter, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		pf, err := NewPriceFetcher(binanceName, &mock.HttpResponseGetterStub{})
+		assert.NotNil(t, pf)
```

Comment:
to make this test a little bit verbose, I would replace all assert.NotNil occurrences with something like:
```GO
assert.Equal(t, "*fetchers.binance", fmt.Sprintf("%T", pf))
```
as you can observe, this works well despite the fact that the factory method returns an unexported instance. Also, this works well between packages.